UI/window-basic-media-controls.cpp

```diff
@@ -0,0 +1,217 @@
+#include "window-basic-media-controls.hpp"
+#include "window-basic-main.hpp"
+
+OBSBasicMediaControls::OBSBasicMediaControls(OBSBasic *parent,
+		OBSSceneItem item_)
+	: QDialog (parent),
+	  main    (parent),
+	  item    (item_)
+{
+	if (item)
+		source = obs_sceneitem_get_source(item);
+
+	isPlaying = false;
+
+	QVBoxLayout *vLayout = new QVBoxLayout();
+	QHBoxLayout *layout = new QHBoxLayout();
+	QHBoxLayout *topLayout = new QHBoxLayout();
+
+	uint32_t seconds = obs_media_get_duration(source) / 1000;
+
+	timerLabel = new QLabel(FormatSeconds(0));
+	durationLabel = new QLabel(FormatSeconds(seconds));
+	playPauseButton = new QPushButton();
+	QPushButton *stopButton = new QPushButton();
+	stopButton->setProperty("themeID", "stopIcon");
+	QPushButton *nextButton = new QPushButton();
+	nextButton->setProperty("themeID", "nextIcon");
+	QPushButton *previousButton = new QPushButton();
+	previousButton->setProperty("themeID", "previousIcon");
+	s = new QSlider();
+	s->setOrientation(Qt::Horizontal);
+	s->setMinimum(0);
+	s->setMaximum(100);
+
+	playPauseButton->setMaximumSize(40, 40);
+	stopButton->setMaximumSize(40, 40);
+	nextButton->setMaximumSize(40, 40);
+	previousButton->setMaximumSize(40, 40);
+
+	topLayout->addWidget(timerLabel);
+	topLayout->addWidget(s);
+	topLayout->addWidget(durationLabel);
+
+	layout->addStretch();
+	layout->addWidget(previousButton);
+	layout->addWidget(playPauseButton);
+	layout->addWidget(stopButton);
+	layout->addWidget(nextButton);
+	layout->addStretch();
+
+	vLayout->addLayout(topLayout);
+	vLayout->addLayout(layout);
+	vLayout->addStretch();
+
+	setLayout(vLayout);
+
+	timer = new QTimer();
+
+	connect(playPauseButton, SIGNAL(clicked()), this,
+			SLOT(MediaPlayPause()));
+	connect(stopButton, SIGNAL(clicked()), this,
+			SLOT(MediaStop()));
+	connect(nextButton, SIGNAL(clicked()), this,
+			SLOT(MediaNext()));
+	connect(previousButton, SIGNAL(clicked()), this,
+			SLOT(MediaPrevious()));
+	connect(s, SIGNAL(sliderMoved(int)),
+			this, SLOT(MediaSliderChanged(int)));
+	connect(s, SIGNAL(sliderReleased()),
+			this, SLOT(MediaSliderReleased()));
+	connect(timer, SIGNAL(timeout()), this, SLOT(Update()));
+
+	uint32_t elapsedTime = obs_media_get_time(source) / 1000;
+	timerLabel->setText(FormatSeconds(elapsedTime));
+
+	float sliderPosition = (((float)elapsedTime * 1000.0f) /
+			obs_media_get_duration(source)) * 100.0f;
+	s->setValue((int)sliderPosition);
+
+	if (obs_media_get_state(source) == OBS_MEDIA_STATE_PLAYING) {
```

Comment:
Instead of calling `obs_media_get_state(source)` repeatedly, consider either a `switch` statement, where the condition is evaluated once, or storing the result of `obs_media_get_state(source)` in a variable and then using that in the `if` condition checks.